tests/tests/recovery.rs

```diff
@@ -203,6 +210,72 @@ proptest! {
     }
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn gas_price_updater__if_no_metadata_history_start_from_current_block(
+) -> anyhow::Result<()> {
+    let driver = FuelCoreDriver::spawn(&[
+        "--debug",
+        "--poa-instant",
+        "true",
+        "--state-rewind-duration",
+        "7d",
+    ])
+    .await?;
+
+    // Given
+    let height = 100;
+    let lower_height = 0;
+    driver.client.produce_blocks(height, None).await?;
+    let database = &driver.node.shared.database;
+    assert_eq!(
+        database.on_chain().latest_height()?,
+        Some(BlockHeight::new(height))
+    );
+
+    let diff = height - lower_height;
+    for _ in 0..diff {
+        let _ = database.gas_price().rollback_last_block();
+    }
+    assert!(database.on_chain().latest_height()? > database.gas_price().latest_height()?);
+    let temp_dir = driver.kill().await;
+
+    // When
+    let recovered_driver = FuelCoreDriver::spawn_with_directory(
+        temp_dir,
+        &[
+            "--debug",
+            "--poa-instant",
+            "true",
+            "--state-rewind-duration",
+            "7d",
+        ],
+    )
+    .await?;
+
+    // Then
+    // advance the block height to the next block to add the metadata to db
+    recovered_driver.client.produce_blocks(1, None).await?;
+    let recovered_database = &recovered_driver.node.shared.database;
+    let next_height = height + 1;
+    assert_eq!(
+        recovered_database.on_chain().latest_height()?,
+        Some(BlockHeight::new(next_height))
+    );
+    assert_eq!(
+        recovered_database.gas_price().latest_height()?,
+        Some(BlockHeight::new(next_height))
+    );
+    let view = recovered_database.gas_price().latest_view().unwrap();
+    let previous_metadata = view
```

Comment:
Why is previous metadata expected to be none?

Reply:
Because there shouldn't be any entries before the one we just added.